code/unclassified/src/running_median_heap/running_median_heap.cpp

```diff
@@ -0,0 +1,369 @@
+/* 
+* Algorithm : Running Median algorithm using Heaps
+*
+* Souce:  
+* https://www.geeksforgeeks.org/median-of-stream-of-integers-running-integers/ 
+*
+* Problem : Given that integers are read from a data stream. 
+* Find median of elements read so for in efficient way. 
+* For simplicity assume there are no duplicates. For example, 
+* let us consider the stream 5, 15, 1, 3 â€¦
+* 
+* After reading 1st element of stream - 5 -> median - 5
+* After reading 2nd element of stream - 5, 15 -> median - 10
+* After reading 3rd element of stream - 5, 15, 1 -> median - 5
+* After reading 4th element of stream - 5, 15, 1, 3 -> median - 4, so on...
+* 
+* when the input size is odd, we take the middle element of sorted data. 
+* If the input size is even, we pick average of middle two elements in 
+* sorted stream.
+*
+* Time Complexity: If we omit the way how stream was read, 
+* complexity of median finding is O(N log N), as we need to read the stream, 
+* and due to heap insertions/deletions.
+*/
+
+#include <iostream>
+
+using namespace std;
+
+// Heap capacity
+#define MAX_HEAP_SIZE (128)
+#define ARRAY_SIZE(a) sizeof(a)/sizeof(a[0])
+
+// Utility functions
+
+// exchange a and b
+void exch(int &a, int &b)
+{
+    int aux = a;
+    a = b;
+    b = aux;
+}
+
+// greaterCompare and smaller are used as comparators
+bool greaterCompare(int a, int b)
+{
+    return a > b;
+}
+
+bool smaller(int a, int b)
+{
+    return a < b;
+}
+
+float average(int a, int b)
+{
+    return (a + b) / 2.0;
+}
+
+/*
+* Signum function
+* = 0  if a == b  - heaps are balanced
+* = -1 if a < b   - left contains less elements than right
+* = 1  if a > b   - left contains more elements than right
+*/
+int signum(int a, int b)
+{
+    if (a == b)
+        return 0;
+
+    return a < b ? -1 : 1;
+}
+
+/*
+* Heap implementation
+* The functionality is embedded into Heap abstract class to avoid 
+* code duplication
+*/
+class Heap
+{
+
+public:
+    // Initializes heap array and comparator require in heapification
+    Heap(int *b, bool (*c)(int, int)) : A(b), comp(c)
+    { 
+        heapSize = -1;
+    }
+
+    // Frees up dynamic memory
+    virtual ~Heap()
+    {
+        if (A)
```

Comment:
If you want to avoid the hassle of dynamic array management use std::array. If y want a dynamic array use std::vector